gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,191 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
+   */
+  Page<Zone> listZones(ZoneListOption... options);
+
+  /**
+   * Deletes an existing zone identified by name. Returns true if the zone was successfully deleted
+   * and false otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and false otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(String zoneName); // delete does not admit any options
+
+  /**
+   * Deletes an existing zone identified by id. Returns true if the zone was successfully deleted
+   * and false otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and false otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(BigInteger zoneId); // delete does not admit any options
+
+  /**
+   * Lists the DNS records in the zone identified by name.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@code
+   * DnsRecordOptions}. Returns null if the zone cannot be found.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(String zoneName, DnsRecordListOption... options);
+
+  /**
+   * Lists the DNS records in the zone identified by ID.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@code
+   * DnsRecordOptions}. Returns null if the zone cannot be found.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(BigInteger zoneId, DnsRecordListOption... options);
+
+  /**
+   * Retrieves the metadata about the current project. The returned fields can be optionally
+   * restricted by specifying {@code ProjectOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects/get">Cloud DNS Projects: get</a>
+   */
+  ProjectInfo getProjectInfo(ProjectGetOption... fields);
+
+  /**
+   * Returns the current project id.
+   */
+  String getProjectId();
+
+  /**
+   * Returns the current project number.
+   */
+  BigInteger getProjectNumber();
+
+  /**
+   * Submits a change requests for applying to the zone identified by ID to the service. The
+   * returned object contains the following read-only fields supplied by the server: id, start time
+   * and status. time, id, and list of name servers. The returned fields can be modified by {@code
+   * ChangeRequestFieldOptions}. Returns null if the zone is not found.
+   *
+   * @return ChangeRequest object representing the new change request or null if zone is not found
```

Comment:
Is ChangeRequest needed? If yes I would use `{@code }` or `{@link}`. "null" -> `{@code null}`